src/hdmint/wallet.cpp

```diff
@@ -315,27 +313,33 @@ bool CHDMintWallet::SeedToZerocoin(const uint512& seedZerocoin, GroupElement& co
     return true;
 }
 
-CKeyID CHDMintWallet::GetZerocoinSeedData(int32_t nCount, uint160 hashSeedMaster){
+CKeyID CHDMintWallet::GetZerocoinSeedData(int32_t nCount){
     // Get CKeyID for n from mintpool
     uint256 hashPubcoin;
     std::pair<uint256,MintPoolEntry> mintPoolEntryPair;
 
-    while(!mintPool.Get(nCount, hashSeedMaster, mintPoolEntryPair)){
+    if(!mintPool.Get(nCount, hashSeedMaster, mintPoolEntryPair)){
         // Top up mintpool if empty
         GenerateMintPool();
+        if(!mintPool.Get(nCount, hashSeedMaster, mintPoolEntryPair)){
+            ResetCount();
+            throw ZerocoinException("Unable to retrieve mint seed ID");
+        }
```

Comment:
Another issue, seem like `GenerateMintPool` generate only next 20 mints if current state is 0 but we need to find at index 1000 it'll never be find. I think we should pass required nCount to function `GenerateMintPool`.

Reply:
Updated following discussion